demisto_sdk/commands/common/tools.py

```diff
@@ -2835,10 +2840,17 @@ def write_dict(
     Write unicode content into a json/yml file.
     """
     path = Path(path)
-    file_handler = handler or (json if path.suffix.lower() == ".json" else yaml)
+    if not handler:
+        suffix = path.suffix.lower()
+        if suffix == ".json":
+            handler = json
+        elif suffix in {".yaml", ".yml"}:
+            handler = yaml
+        else:
+            raise ValueError(f"The file {path} is neither json/yml")
 
     safe_write_unicode(
-        lambda f: file_handler.dump(data, f, indent, sort_keys, **kwargs), path
+        lambda f: handler.dump(data, f, indent, sort_keys, **kwargs), path  # type: ignore[union-attr]
```

Comment:
why the type-ignore?

Reply:
looks like mypy doesn't recognize that handler will not be `None` in any situation, perhaps its going by the function signature?